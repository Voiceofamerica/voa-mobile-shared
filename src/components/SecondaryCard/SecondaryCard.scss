
@import '../../commonStyles/index.scss';

.card {
  @include touchable();

  flex: 1;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  overflow: hidden;
  -webkit-overflow-scrolling: touch;
  user-select: none;

  .imageContainer {
    height: 70%;
  }

  .titleText {
    height: 30%;
    overflow: hidden;
    font-weight: bold;
  }

  & + .card {
    margin-left: 1.5vw;

    [dir="rtl"] & {
      margin-left: 0;
      margin-right: 1.5vw;
    }
  }
}
