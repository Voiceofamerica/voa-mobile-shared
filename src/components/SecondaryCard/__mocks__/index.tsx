
import * as React from 'react'

export default (props) => (
  React.createElement('secondary-card', { props }, props.children)
)

export const getImageHeight = () => {
  return 10
}

export const getHeight = () => {
  return 90
}
