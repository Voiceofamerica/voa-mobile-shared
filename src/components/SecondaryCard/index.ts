
export { default } from './SecondaryCard'
export * from './SecondaryCard'
