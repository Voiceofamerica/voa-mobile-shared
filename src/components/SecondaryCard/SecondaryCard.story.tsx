
import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import ArticleBlurb from '../../types/ArticleBlurb'

import SecondaryCard from './SecondaryCard'

const testArticle = {
  imageUrl: 'https://gdb.voanews.com/3E1C440D-1C78-4B88-84F7-ECB12FB5B874_cx3_cy0_cw97_w650_r1.png',
  minorText: '15:08',
  title: 'President Trump: From North Korea to Iran, we confront dangerous governments',
}

const testArticle2 = {
  imageUrl: 'https://gdb.voanews.com/01D5530E-3642-4CBF-957E-AE990ABE9167_cx0_cy14_cw0_w650_r1.jpg',
  pubDate: '21:55',
  title: 'At least seven people were killed in a police vehicle in Balochistan, Pakistan',
}

storiesOf('SecondaryCard', module)
  .add('display two', () => (
    <div style={{ display: 'flex', flexDirection: 'row' }}>
      <SecondaryCard onPress={action('Card.onPress')} { ...testArticle } />
      <SecondaryCard onPress={action('Card.onPress')} { ...testArticle2 } />
    </div>
  ))
  .add('rtl display two', () => (
    <div style={{ display: 'flex', flexDirection: 'row' }} dir='rtl'>
      <SecondaryCard onPress={action('Card.onPress')} { ...testArticle } />
      <SecondaryCard onPress={action('Card.onPress')} { ...testArticle2 } />
    </div>
  ))
