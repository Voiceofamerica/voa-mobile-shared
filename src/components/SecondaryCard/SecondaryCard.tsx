
import * as React from 'react'

import ResilientImage from '../ResilientImage'

import { card, imageContainer, titleText } from './SecondaryCard.scss'

export interface Props {
  title: JSX.Element|string
  onPress?: () => void
  imageUrl?: string
  className?: string
  style?: React.CSSProperties
}

class SecondaryCard extends React.Component<Props> {
  state = {
    windowWidth: window.innerWidth,
  }

  render () {
    const { onPress = () => null, title, imageUrl, className, style } = this.props

    return (
      <div
        className={`${card} ${className}`}
        style={style}
        onClick={onPress}>
        <ResilientImage className={imageContainer} src={imageUrl} alwaysShow />
        <div className={titleText}>
          {title}
        </div>
      </div>
    )
  }
}

export default SecondaryCard
