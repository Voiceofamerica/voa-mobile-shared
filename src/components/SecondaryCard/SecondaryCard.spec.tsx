
import * as React from 'react'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

import SecondaryCard from './SecondaryCard'

jest.mock('../ResilientImage')

describe('<SecondaryCard />', () => {
  describe('shapshots', () => {
    it('should render with the given title', () => {
      const element = create((
        <SecondaryCard title='myTitle' />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given image', () => {
      const element = create((
        <SecondaryCard title='myTitle' imageUrl='myImage' />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given className', () => {
      const element = create((
        <SecondaryCard title='myTitle' className='myClass' />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given style', () => {
      const element = create((
        <SecondaryCard title='myTitle' style={{ top: 0, left: 0, right: 0, bottom: 0 }} />
      ))

      expect(element).toMatchSnapshot()
    })
  })

  describe('onPress', () => {
    it('should be called whenever the element is clicked', () => {
      const onPress = jest.fn()

      const item = shallow((
        <SecondaryCard title='myTitle' onPress={onPress} />
      ))

      item.simulate('click')

      expect(onPress).toHaveBeenCalledTimes(1)
    })

    it('should not throw an error if not provided and the element is clicked', () => {
      const item = shallow((
        <SecondaryCard title='myTitle' />
      ))

      item.simulate('click')
    })
  })
})
