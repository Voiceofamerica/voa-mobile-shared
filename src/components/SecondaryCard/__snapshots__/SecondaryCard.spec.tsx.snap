// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondaryCard /> shapshots should render with the given className 1`] = `
<div
  className="card myClass"
  onClick={[Function]}
  style={undefined}
>
  <div
    alwaysShow={true}
    className="imageContainer"
    src={undefined}
  />
  <div
    className="titleText"
  >
    myTitle
  </div>
</div>
`;

exports[`<SecondaryCard /> shapshots should render with the given image 1`] = `
<div
  className="card undefined"
  onClick={[Function]}
  style={undefined}
>
  <div
    alwaysShow={true}
    className="imageContainer"
    src="myImage"
  />
  <div
    className="titleText"
  >
    myTitle
  </div>
</div>
`;

exports[`<SecondaryCard /> shapshots should render with the given style 1`] = `
<div
  className="card undefined"
  onClick={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    alwaysShow={true}
    className="imageContainer"
    src={undefined}
  />
  <div
    className="titleText"
  >
    myTitle
  </div>
</div>
`;

exports[`<SecondaryCard /> shapshots should render with the given title 1`] = `
<div
  className="card undefined"
  onClick={[Function]}
  style={undefined}
>
  <div
    alwaysShow={true}
    className="imageContainer"
    src={undefined}
  />
  <div
    className="titleText"
  >
    myTitle
  </div>
</div>
`;
